src/checks/kubernetes/services/service.go

```diff
@@ -88,28 +88,67 @@ func (i inputs) GeneralCheck() Results {
 			panic(err.Error())
 		}
 
-		if values.Values.ChecksEnabled.ClusterIP {
-			fmt.Println("fooooooo")
-		}
-
-		if values.Values.ChecksEnabled.Endpoints {
-			fmt.Println("fooooooo")
-		}
+		// Loop through all of the services found
+		for _, aService := range services.Items {
+
+			// Find the service with the name we are interested in
+			if aService.ObjectMeta.Name == values.Values.ServiceName {
+				//
+				// Run only enabled checks
+				//
+
+				if values.Values.ChecksEnabled.ClusterIP {
+					if aService.Spec.ClusterIP == "" {
+						checkResult.DidPass = false
+						checkResult.Message += "* No ClusterIP Found\n"
+					} else {
+						checkResult.Message += "* ClusterIP Found\n"
+					}
+				}
 
-		if values.Values.ChecksEnabled.HostPort {
-			fmt.Println("fooooooo")
-		}
+				if values.Values.ChecksEnabled.Endpoints {
+					// Look to see if the endpoints for this service exists or not
+					endpoints, err := kubeClientSet.CoreV1().Endpoints(i.namespace).List(context.TODO(), metav1.ListOptions{})
+					if err != nil {
+						panic(err.Error())
+					}
 
-		if values.Values.ChecksEnabled.Ports {
-			for _, aService := range services.Items {
+					for _, anEndpoint := range endpoints.Items {
```

Comment:
There seems to be an overwhelming amount of iterating and conditional testing here. It is acceptable to me if this is a first-pass effort, though it is hard for me to follow the nesting in earnest.

Reply:
what do you suggest that we do here instead?